src/base/action.lua

```diff
@@ -257,6 +257,47 @@
 		return false
 	end
 
+---
+-- Determines if an action supports a particular toolset.
+--
+-- @param language
+--    The language that toolset belongs.
+-- @param toolset
+--    The toolset to check.
+-- @returns
+--    True if the toolset is supported, false otherwise.
+---
+
+	function action.supportsToolset(language, toolset)
+		local function buildLanguageKeysMap()
+			local language_keys_map = {C = "cc", D = "dc" }
+			language_keys_map["C++"] = "cc"
+			language_keys_map["C#"] = "dotnet"
+			return language_keys_map
+		end
+		if not language or not toolset then
+			return true
+		end
```

Comment:
Can you comment the logic here? If neither is specified, why does it return true?

Can you do the same for the other `if not <thing> then` blocks that return true?

Reply:
As above function in fact. we pass `prj.language`, `prj.toolset` which might be not provided, so `nil`.
So handle that case.
For unknown/unsupported languages (rust, F#), it is just to be future proof and avoid warning.